arangod/Aql/OptimizerRules.cpp

```diff
@@ -2655,55 +2648,50 @@ void arangodb::aql::removeRedundantCalculationsRule(
 
       if (current->getType() == EN::CALCULATION) {
         try {
+          buffer.clear();
           // ExecutionNode::castTo<CalculationNode*>(current)->expression()->node()->dump(0);
```

Comment:
remove?

Reply:
I think the `clear()` should stay in place here, because otherwise `buffer` is not guaranteed to be empty. It is populated one or two lines below by the call to `Expression::stringifyIfNotTooLong(...)`, but that only appends to the buffer and doesn't clear it initially.
And this stuff is executed in a loop. My conclusion is: clear() must still be there to keep the same behavior.

Reply:
sorry I meant the comment 1 line below :-)
```
// ExecutionNode::castTo<CalculationNode*>(current)->expression()->node()->dump(0);
```